MCLOUD-6977: Update documentation about routes (#7912)
magento/devdocs: src/cloud/project/routes.md

 - /cloud/project/project-conf-files_routes.html  
---

The `routes.yaml` file in the `.magento/routes.yaml` directory defines routes for your {{ site.data.var.ece }} Integration, Staging, and Production environments. Routes determine how Magento processes incoming HTTP and HTTPS requests.

{% include cloud/note-pro-missing-self-service-options.md %}

> The default `routes.yaml` file specifies the route templates for processing as HTTP as HTTPS requests on projects that have a single default domain and on projects configured for multiple domains:

Reviewer: ```suggestion
The default `routes.yaml` file specifies the route templates for processing HTTP requests as HTTPS on projects that have a single default domain and on projects configured for multiple domains:
```

{:.no-copy}

For example, the `refactorcss` branch for the `mswy7hzcuhcjw` project hosted in the `us` cluster has the following the domain:

```text
https://refactorcss-oy3m2pq-mswy7hzcuhcjw.us.magentosite.cloud/
```
  {:.no-copy}

{:.bs-callout-info}
If your Cloud project supports multiple stores, follow the route configuration instructions for [multiple websites or stores]({{ site.baseurl }}/cloud/project/project-multi-sites.html).

## Route protocols

All environments support both HTTP and HTTPS automatically.

- If the configuration specifies only the HTTP route, HTTPS routes are created automatically, allowing the site to be served from both HTTP and HTTPS without requiring redirects. For example, if you have project with the default domain `example.com`, the record `http://{default}/` resolves to the following URLs:

   ```text
   http://example.com/

   https://example.com/
   ```
   {:.no-copy}

- If the configuration specifies only the HTTPS route, then all HTTP requests redirect to HTTPS. For example, for the default domain `example.com`, the route `https://{default}/` resolves to URL `https://example.com/` and redirects `http://example.com/` to `https://example.com/`.

> To serve all pages over TLS is recommended.
> For this you need to configure redirects of all unencrypted request to the TLS equivalent. You can do it in one of the following ways:
> 

Reviewer: Nice addition of the recommendation and different methods.

```suggestion
We recommend serving all pages over TLS.  For this configuration, you must configure redirects for all unencrypted request to the TLS equivalent using one of the following methods:

```

- Change protocol to HTTPS in `routes.yaml`.

```yaml
"https://{default}/":
    type: upstream
    upstream: "mymagento:http"
"https://{all}/":
    type: upstream
    upstream: "mymagento:http"
```

> - For Staging and Production environments we recommend to enable [Force TLS on Fastly](https://support.magento.com/hc/en-us/articles/360006296953-Redirect-HTTP-to-HTTPS-for-all-pages-on-Cloud-Force-TLS-) without changes in `routes.yaml`.

Reviewer: ```suggestion
- For Staging and Production environments, we recommend enabling the [Force TLS on Fastly](https://support.magento.com/hc/en-us/articles/360006296953-Redirect-HTTP-to-HTTPS-for-all-pages-on-Cloud-Force-TLS-) option from the Magento Admin UI.  When you use this option, Fastly handles the redirection to HTTPS, so you do not have to update the `routes.yaml` configuration for your Cloud Staging and Production environments.
```

  {:.no-copy}

{:.bs-callout-info}

> Follow the instruction to configure [multiple websites or stores]({{ site.baseurl }}/cloud/project/project-multi-sites.html) on your Cloud project.

Reviewer: ```suggestion
If your Cloud project supports multiple stores, follow the route configuration instructions for [multiple websites or stores]({{ site.baseurl }}/cloud/project/project-multi-sites.html).
```

## Route protocols

All environments support both HTTP and HTTPS automatically. 

- If only HTTP routes are specifying, HTTPS routes being created automatically, allowing the site to be served from both HTTP and HTTPS without redirects.
For example, if you have project with default domain `example.com`, the record `http://{default}/` will resolve following URLs:

   ```text
   http://example.com/
   
   https://example.com/
   ```
   {:.no-copy}

- If only HTTPS routes are specifying, then all HTTP request will be redirected to HTTPS.
For example, for default domain `example.com`, route `https://{default}/` will create 2 URLs

   ```text
   http://example.com/
   
   https://example.com/
   ```

> But request to `http://example.com/` will always redirect to `https://example.com/`

Reviewer: ```suggestion
However, the request to `http://example.com/` always redirects to `https://example.com/`
```

  {:.no-copy}

{:.bs-callout-info}
If your Cloud project supports multiple stores, follow the route configuration instructions for [multiple websites or stores]({{ site.baseurl }}/cloud/project/project-multi-sites.html).

## Route protocols

All environments support both HTTP and HTTPS automatically.

- If the configuration specifies only the HTTP route, HTTPS routes are created automatically, allowing the site to be served from both HTTP and HTTPS without requiring redirects. For example, if you have project with the default domain `example.com`, the record `http://{default}/` resolves to the following URLs:

   ```text
   http://example.com/

   https://example.com/
   ```
   {:.no-copy}

- If the configuration specifies only the HTTPS route, then all HTTP requests redirect to HTTPS. For example, for the default domain `example.com`, the route `https://{default}/` resolves to URL `https://example.com/` and redirect `http://example.com/` to `https://example.com/`.

> What is similar to

Reviewer 2: ```suggestion
Which is similar to
```

> What is similar to

Reviewer: As discussed, the phrase "What is similar to" and example can be deleted as the preceding bullet point explains the concept.

Author: done

  {:.no-copy}

{:.bs-callout-info}
If your Cloud project supports multiple stores, follow the route configuration instructions for [multiple websites or stores]({{ site.baseurl }}/cloud/project/project-multi-sites.html).

## Route protocols

All environments support both HTTP and HTTPS automatically.

- If the configuration specifies only the HTTP route, HTTPS routes are created automatically, allowing the site to be served from both HTTP and HTTPS without requiring redirects. For example, if you have project with the default domain `example.com`, the record `http://{default}/` resolves to the following URLs:

   ```text
   http://example.com/

   https://example.com/
   ```
   {:.no-copy}

- If the configuration specifies only the HTTPS route, then all HTTP requests redirect to HTTPS.

> For example, for the default domain `example.com`, the route `https://{default}/` resolves to URL `https://example.com/` and redirect `http://example.com/` to `https://example.com/`.

Reviewer: ```suggestion
For example, for the default domain `example.com`, the route `https://{default}/` resolves to URL `https://example.com/` and redirects `http://example.com/` to `https://example.com/`.
```

What is similar to

    ```yaml
    "https://{default}/":
        type: upstream
        upstream: "mymagento:http"

    "http://{default}/":
        type: redirect
        to: "https://{default}/"
    ```

> Where the request to `http://example.com/` always redirects to `https://example.com/`

Reviewer: ```suggestion
The request to `http://example.com/` always redirects to `https://example.com/`
```

## Simple routes

> The following sample routes the apex domain and the `www` subdomain to the `mymagento` application. This route does not redirect HTTPS requests:

Reviewer: I am a little confused by the sentence: "This route does not redirect HTTPS requests" -- Since HTTPS is the standard now, why would we want to provide an example that doesn't redirect HTTPS requests. 

```suggestion
In the following examples, the route configuration routes the apex domain and the `www` subdomain to the `mymagento` application. This route does not redirect HTTPS requests.

**Example 1:**

```

```yaml

> "http://{default}/":

Reviewer: Add Example 2: line to label the second example.  

Change explanatory text to 

"In the following example, the route configuration does not redirect URLs from the `www` domain to the apex domain. Instead, requests are served from both the `www` and apex domain."

Author: done

    type: upstream
    upstream: "mymagento:http"

"http://www.{default}/":
    type: upstream
    upstream: "mymagento:http"
```

> In the first sample, the server responds directly to a request of the form `http://example.com/hello`, but it issues a _301 redirect_ for `http://www.example.com/mypath` (to `http://example.com/mypath`).

Reviewer: Move this explanation to paragraph after Example 1: 
```suggestion
In this example, the server responds directly to a request of the form `http://example.com/hello`, issuing a _301 redirect_ for requests with the URL pattern `http://www.example.com/mypath`. These requests redirect  to the apex domain, for example `http://example.com/mypath`.
```
